ecs.tf

```diff
@@ -13,7 +13,7 @@ resource "aws_ecs_cluster" "the_cluster" {
 }
 
 resource "aws_ecs_capacity_provider" "primary" {
-  name = aws_autoscaling_group.ecs_asg.name
+  name = "cp-${md5(join(var.managed_scaling_maximum_scaling_step_size, var.managed_scaling_maximum_scaling_step_size, var.managed_scaling_target_capacity, var.managed_scaling_status, var.managed_termination_protection))}"
```

Comment:
But why....

Reply:
Because we have to create a unique string that represents the state of the capacity provider. As-is when we re-terraform the provider it uses the ASG name, which is the same name as the previous provider IF and only if the ASG doesn't change. In that case, terraform will forever error because it tries to add a provider that already exists.